### PR TITLE
Add netcore50 and net46 configurations for System.Net.Sockets

### PR DESCRIPTION
**Pull request by mellinoe**
@davidsh , @CIPop , @weshaggard

### CODE REVIEW

File: src/System.Net.Sockets/src/System.Net.Sockets.csproj [hunk 1]
```diff
@@ -38,9 +44,15 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <!-- All configurations -->
+    <Compile Include="System\Net\Sockets\SocketReceiveFromResult.cs" />
+    <Compile Include="System\Net\Sockets\SocketReceiveMessageFromResult.cs" />
+    <Compile Include="System\Net\Sockets\SocketTaskExtensions.cs" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetGroup)' != 'net46'">
```
Comment (CIPop) [inline src/System.Net.Sockets/src/System.Net.Sockets.csproj]: Please add a comment here describing what SKU will use this code? What does it mean _not_ to be net46?

Comment (mellinoe) [inline src/System.Net.Sockets/src/System.Net.Sockets.csproj]: I added a few comments clarifying which platforms some ItemGroup's were applicable to.

File: src/System.Net.Sockets/src/System.Net.Sockets.builds [hunk 2]
```diff
@@ -11,6 +11,13 @@
     <Project Include="System.Net.Sockets.csproj">
       <OSGroup>Windows_NT</OSGroup>
     </Project>
+    <Project Include="System.Net.Sockets.csproj">
+      <TargetGroup>net46</TargetGroup>
+    </Project>
+    <Project Include="System.Net.Sockets.csproj">
+      <TargetGroup>netcore50</TargetGroup>
```
Comment (weshaggard) [inline src/System.Net.Sockets/src/System.Net.Sockets.builds]: Nit: OSGroup above TargetGroup. I'm going to run a script on the repo again soon to make these consistent and to help identify any miss matches between .builds and the configurations in the project files.

**Comment by weshaggard** [comment]
LGTM

**Comment by CIPop** [comment]
LGTM

Thanks @mellinoe!

**Pull request merged**